File: java/src/main/java/de/sven_torben/serialization_benchmark/logging/CombinedLogger.java
package de.sven_torben.serialization_benchmark.logging;

import java.util.Arrays;

public class CombinedLogger implements Logger {

	private final Logger[] loggers;

	public CombinedLogger(final Logger... loggers) {
		this.loggers = loggers;
	}

	@Override
	public void log(final String serializerName, final int bytes,
			final long[] writeTimes, final long[] readTimes) {
		Arrays.stream(loggers).parallel().forEach(logger -> logger.log(serializerName, bytes, writeTimes,
				readTimes));
	}

}

File: net/Logging/CombinedLogger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace De.Sven_Torben.Serialization_Benchmark.Logging
{
    class CombinedLogger : Logger
    {
        private Logger[] loggers;

        public CombinedLogger(params Logger[] loggers)
        {
            this.loggers = loggers;
        }

        public void Log(string serializerName, int bytes, long[] writeTimes, long[] readTimes)
        {
            loggers.AsParallel().ForAll(l => l.Log(serializerName, bytes, writeTimes, readTimes));
        }
    }
}
